modules/http/src/test/java/org/mule/module/http/functional/listener/HttpListenerMalformedUrlTestCase.java

```diff
@@ -0,0 +1,123 @@
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.apache.commons.lang.StringEscapeUtils.escapeHtml;
+import static org.mule.module.http.api.HttpConstants.HttpStatus.BAD_REQUEST;
+import static org.mule.module.http.api.HttpConstants.HttpStatus.OK;
+
+import org.mule.module.http.utils.SocketRequester;
+import org.mule.tck.junit4.FunctionalTestCase;
+import org.mule.tck.junit4.rule.DynamicPort;
+
+import com.google.common.base.Charsets;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+
+import org.apache.http.client.fluent.Request;
+import org.apache.http.HttpResponse;
+import org.apache.http.StatusLine;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mule.util.IOUtils;
+
+public class HttpListenerMalformedUrlTestCase extends FunctionalTestCase
+{
+
+    public static final String TEST_MALFORMED = "api/ping%";
+    public static final String TEST_MALFORMED_SCRIPT = "<script></script>%";
+    public static final String TEST_ENCODED_SPACE = "test/foo 1 %";
+    public static final String TEST_ENCODED_HASHTAG = "test/foo 1 #";
+    public static final String TEST_ENCODED_PERCENT2 = "test/%24";
+
+    @Rule
+    public DynamicPort httpPort = new DynamicPort("httpPort");
+
+    @Override
+    protected String getConfigFile()
+    {
+        return "http-listener-malformed-url-config.xml";
+    }
+
+    @Test
+    public void returnsBadRequestOnMalformedUrlForWildcardEndpoint() throws Exception
+    {
+        SocketRequester socketRequester = new SocketRequester("localhost", httpPort.getNumber());;
+        try
+        {
+            socketRequester.initialize();
+            socketRequester.doRequest("GET /" + TEST_MALFORMED + " HTTP/1.1");
+            String response = socketRequester.getResponse();
+            assertThat(response, containsString(Integer.toString(BAD_REQUEST.getStatusCode())));
+            assertThat(response, containsString(BAD_REQUEST.getReasonPhrase()));
+            assertThat(response, containsString("Unable to parse request: /"+TEST_MALFORMED));
+        }
+        finally
+        {
+            socketRequester.finalizeGracefully();
+        }
+    }
+
+    @Test
+    public void returnsBadRequestOnMalformedUrlWithInvalidContentTypeWithScript() throws Exception
+    {
+        SocketRequester socketRequester = new SocketRequester("localhost", httpPort.getNumber());;
+        try
+        {
+            socketRequester.initialize();
+            socketRequester.doRequest("POST /"+TEST_MALFORMED_SCRIPT+" HTTP/1.1");
+            String response = socketRequester.getResponse();
+            assertThat(response, containsString(Integer.toString(BAD_REQUEST.getStatusCode())));
+            assertThat(response, containsString(BAD_REQUEST.getReasonPhrase()));
+            assertThat(response, containsString(escapeHtml(TEST_MALFORMED_SCRIPT)));
```

Comment:
For these body assertions we should be specific, maybe using `endsWith` and considering the new line before the body.